core/primitives/src/merkle.rs

```diff
@@ -98,6 +98,41 @@ pub fn verify_path<T: BorshSerialize>(root: MerkleHash, path: &MerklePath, item:
     hash == root
 }
 
+#[derive(Default, Clone, BorshSerialize, BorshDeserialize)]
```

Comment:
Could we add explanation of what this type represents, e.g. what is "next leaf" and why number of leaves is important, and why is it called "size"?

Reply:
Okay